### PR TITLE
Added code for Regular Expression Matching

### PR DESCRIPTION
**Pull request by Vinaysukhwal**
### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: dynamic_programming/regular_expression_matching.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+"""
+Given an input string (s) and a pattern (p), implement regular expression matching with support for '.' and '*' where: 
+
+'.' Matches any single character.
+'*' Matches zero or more of the preceding element.
+The matching should cover the entire input string (not partial).
+
+https://leetcode.com/problems/regular-expression-matching/
+"""
+def isMatch(self, s: str, p: str) -> bool:
```
Comment (mrmaxguns) [inline dynamic_programming/regular_expression_matching.py:10]: Python naming conventions state that function names should be in snake case and one letter variable names are not acceptable. Also, there is no need for a `self` argument as the function isn't a class method:

```python
def is_match(check_string: str, pattern: str) -> bool:
    ...
```

File: dynamic_programming/regular_expression_matching.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        dp = [[0 for i in range(len(p)+1)] for j in range(len(s)+1)]
```
Comment (mrmaxguns) [inline dynamic_programming/regular_expression_matching.py:11]: What is `dp`? Variable names should be short, but descriptive.

File: dynamic_programming/regular_expression_matching.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        
+        for j in range(2,len(p)+1):
+            if(p[j-1] == "*"):
+                dp[0][j] = dp[0][j-2]
+                
+        
+        
+        for i in range(1,len(s)+1):
+            for j in range(1,len(p)+1):
+                if(p[j-1]==s[i-1] or p[j-1]=="."):
```
Comment (mrmaxguns) [inline dynamic_programming/regular_expression_matching.py:23]: The parentheses are unnecessary and redundant:

```python
if p[j - 1] == s[i - 1] or p[j - 1] == ".":
    ...
```

File: dynamic_programming/regular_expression_matching.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                    dp[i][j] = dp[i-1][j-1]
+                    
+                elif(j>1 and p[j-1]=="*"):
```
Comment (mrmaxguns) [inline dynamic_programming/regular_expression_matching.py:26]: These parentheses can be removed.

**Review by mrmaxguns — CHANGES_REQUESTED** [review]

**Comment by xcodz-dot** [comment]
### This PR might get closed within 4 days from now:
* No doctests
* Pre-commit failed
* No author response

### Fix:
* Give a comment out here so we know you are here.
* Make required changes

You can ask me if you need to

**Pull request closed without merging**